### PR TITLE
Add Dockerfile pt. 2

### LINKED ISSUE #101: [FEATURE REQ] Add Dockerfile for `grofer`
**Describe the solution you'd like**
Have a Dockerfile for grofer for ease of use. 
```
docker run pesos/grofer:latest
```

**Additional context**
- On initial thought, this can be done by mounting the file systems being used by `gopsutil` to fetch metrics, `/proc` and `/sys` might be a good starting point.

### PR DESCRIPTION
**Pull request by Samyak2**
# Description

Fixes #101 

Based on #106 

 - Added a Dockerfile based on the [`golang` image](https://hub.docker.com/_/golang)
 - `grofer` is installed using `go install` from the local source
 - The whole root directory of the host has to be mounted as read only for this to work. A few other docker flags are also required.
    - the docker socket is also mounted
 - See updated README for instructions
 - `HOST_PROC`, `HOST_SYS`, etc. is set for `gopsutil`

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [x] I have read the [contribution guidelines](https://github.com/pesos/grofer/blob/master/CONTRIBUTING.md) and followed it as far as possible. 
- [x] I have performed a self-review of my own code (if applicable)
- [x] I have commented my code, particularly in hard-to-understand areas (if applicable)
- [x] I have run `go fmt` on my code ([reference](https://blog.golang.org/gofmt))
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] My changes generate no new warnings 
- [x] Any dependent and pending changes have been merged and published

# TODO

 - fix GitHub workflow to publish container

### CODE REVIEW

**Comment by Samyak2** [comment]
```
unsupported: GitHub Container Registry is currently in read only mode. Image pushes are disallowed at this time.
```

Looks like the container registry is [under maintenance](https://www.githubstatus.com/) currently :(

**Comment by Samyak2** [comment]
It wooooorks!

https://github.com/orgs/pesos/packages/container/package/grofer

**Comment by MadhavJivrajani** [comment]
WOOOHOOOOOOO

File: README.md
Comment (MadhavJivrajani) [inline README.md:51]: Now I guess we can fill this in 🎉

Comment (Samyak2) [reply README.md:51]: Yes, I'm doing that rn.

I can't exactly test it since `latest` isn't pushed yet. Pulling using `pr-124` tag works though.

**Comment by Samyak2** [comment]
I think this is ready now. I made a few changes
 - Docker workflow is disabled for PRs since it will fail unless it's from an internal branch (like this)
 - Changed the README to use `ghcr.io/pesos/grofer` instead of `grofer`

**Comment by Gituser143** [comment]
@Samyak2 you might want to update the branch :P

**Comment by Samyak2** [comment]
I'll rebase it on main

**Comment by Samyak2** [comment]
Is there anything left to do here?

**Comment by MadhavJivrajani** [comment]
Everything looks good to me, just to be sure: this image build will be triggered only when a new release comes out?

**Comment by Samyak2** [comment]
No. This is how it will work:
- All pushes to main (the default branch) are released as the `latest` image.
- New tags are released with the specific version number. For example, a `1.3.1` tagged image would be pushed on the next release.

So, pulling `ghcr.io/pesos/grofer` would give the image from the main branch (since that is tagged as `latest`). This, if I'm not wrong, is similar to `go get` which pulls from the default branch.

**Comment by MadhavJivrajani** [comment]
Sounds good
Could you add 1-2 lines in the READE along the lines of what `latest` means in our case and if they want to use a particular version, how they can do it?

**Comment by Samyak2** [comment]
I've added a couple of lines on that.

A few things that should be noted here:
 - The `latest` image and release images (tags) are not tested xD. They can only be tested once we merge this (and the tag one when we make a new release). `act` can't test these either since it fails at the login step. We might have to make hotfixes if it breaks. On that note, is it possible to force Actions to run the release (tag) workflow manually so that the `1.3.0` image is built?
 - I'm not entirely sure about the version tag. According to the code (which I ~stole~ borrowed from the [starter workflows](https://github.com/actions/starter-workflows/blob/68d67073c3864b2836a59ed0af5b30d2ad3ff719/ci/docker-publish.yml#L71)) it should be `1.3.0` without the `v` prefix. We might have to change this in the README if it's different.

**Comment by Samyak2** [comment]
^ Rebased on main

Is there anything else to do here?

**Review by MadhavJivrajani — APPROVED** [review]
Hey, I missed this
I see there are a few small details regarding version to be worked out, but that can be done independently later, with the documentation in the README, this looks good to me in the current state and can be merged

cc @Gituser143

**Review by Gituser143 — APPROVED** [review]

**Comment by Gituser143** [comment]
@Samyak2 
Changes look good to me! But do workflows trigger on commits? I'm asking because I don't see one for this (or I might not be looking properly, plis help :P).

**Comment by Samyak2** [comment]
> @Samyak2 
> Changes look good to me! But do workflows trigger on commits? I'm asking because I don't see one for this (or I might not be looking properly, plis help :P).

@Gituser143 yes, but this one is run only on commits to the main branch and not on PRs. It can be enabled on PRs too, but if they aren't from an internal branch of this repo (pesos/grofer), it will just give a permission denied error.
Also, it's run on push rather than commits. So if multiple commits are pushed at a time, it will run only once for the latest commit.

**Pull request merged**
